SU-HSDO/suhumsci
# SHS-5775: Implementation: Social Media Footer Block

## Summary
Adds config and update hook to create new social media footer block.

## Need Review By (Date)
2024-09-09

## Urgency
low

## Steps to Test
1. Login to the site as a Developer role
2. Under the Content menu, hover over "Blocks", then "Add content block", then click "Social Media Footer Block"

![Screenshot 2024-09-05 at 1 19 07 PM](https://github.com/user-attachments/assets/809c323b-2307-4f7e-93ff-87b48c976422)

4. Fill out the fields for description, icon size, layout, and add as many links as you wish
5. Click "Save and configure" at the bottom
6. When on the "Configure block" screen, select "Footer" as the value for the "Region" field
7. At the bottom, click "Save block"
8. View the front-end of the site and scroll all the way to the footer area
9. Verify that there is a new social media footer block present, with the correct styles according to the selected icon size and layout

Grid layout:
<img width="245" alt="Screenshot 2024-09-23 at 10 07 15 AM" src="https://github.com/user-attachments/assets/eeb686b7-d02c-43ac-ab8d-91cdc7787195">

List layout:
<img width="294" alt="Screenshot 2024-09-23 at 10 07 48 AM" src="https://github.com/user-attachments/assets/518952d3-44ac-4c23-a8b6-eb46606fa35e">

10. Masquerade as a user who does not have a developer role, for example someone who only has "Site manager" role
11. View the front-end of the site and scroll all the way to the footer area
12. Hover over the social media block and click the pencil icon
13. Verify that the only thing a non-developer role can do is edit the block, not remove it or configure it.

<img width="260" alt="Screenshot 2024-09-23 at 10 08 54 AM" src="https://github.com/user-attachments/assets/538d74b0-f4c5-4e4e-ace2-54026fa40ca3">

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
